### PR TITLE
pua_reginfo: free pkg memory after getting the XML attribute

### PR DESCRIPTION
**Pull request by apogrebennyk**
fix(pua_reginfo): free pkg memory after getting the XML attribute
- free the memory from the state XML attribute that was returned and immediately used in another function call without freeing leaving one small mem leak;
- add comments but keep the dead code regarding parsed but unused attrs.

#### Pre-Submission Checklist

- [x] Commit message has the format required by CONTRIBUTING guide
- [x] Commits are split per component (core, individual modules, libs, utils, ...)
- [x] Each component has a single commit (if not, squash them into one commit)
- [x] No commits to README files for modules (changes must be done to docbook files
in `doc/` subfolder, the README file is autogenerated)

#### Type Of Change
- [x] Small bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)

#### Checklist:

- [x] PR should be backported to stable branches
- [x] Tested changes locally
- [ ] Related to issue #XXXX (replace XXXX with an open issue number)

#### Description
After testing that this change brings positive impact I found similar fix that has een done by Jason Penton in modules/ims_registrar_pcscf: fix mem leak (libxml2 structure not being freed)
cedae42f jaybeepee <jason.penton@gmail.com> on 9/28/16 at 12:26 PM
After getting content of each XML attribute that returns dynamic memory the memory should be de-allocated.

### CODE REVIEW

**Comment by apogrebennyk** [comment]
Local test running, I will update the checklist once I notice no new memory leaks.

**Comment by apogrebennyk** [comment]
This looks stable now.

File: src/modules/pua_reginfo/notify.c [hunk 1]
```diff
@@ -417,12 +450,17 @@ int process_body(str notify_body, udomain_t *domain)
 					if(xmlStrcasecmp(params->name, BAD_CAST "unknown-param")
 							!= 0)
 						goto next_param;
-					len += 1 /* ; */
-						   + strlen(xmlGetAttrContentByName(params, "name"));
+					char *param_name = xmlGetAttrContentByName(params, "name");
```
Comment (linuxmaniac) [inline src/modules/pua_reginfo/notify.c:453]: char *param_name should be defined at the top

File: src/modules/pua_reginfo/notify.c [hunk 2]
```diff
@@ -439,10 +477,11 @@ int process_body(str notify_body, udomain_t *domain)
 						if(xmlStrcasecmp(params->name, BAD_CAST "unknown-param")
 								!= 0)
 							goto next_param2;
+						char *param_name =
```
Comment (linuxmaniac) [inline src/modules/pua_reginfo/notify.c:480]: just one definition?

**Comment by apogrebennyk** [comment]
@linuxmaniac the PR comments have been incorporated, please recheck.

File: src/modules/pua_reginfo/notify.c (hunk 1 above)
Comment (apogrebennyk) [reply src/modules/pua_reginfo/notify.c]: done, thanks

**Pull request merged**